packages/backend/src/payment-method/ilp/service.ts

```diff
@@ -217,12 +217,24 @@ async function pay(
     })
   }
 
+  // TODO: prevent ilpQuoteDetails with better type instead of having to check and error?
+  // Or make better error.
+  // Is there a way to ensure that right here, outgoingPayment.quote is QuoteWithDetails?
+  // I dont think so... this is where the ambiguousness of our DB types surfaces and we have
+  // to do some explicit check like this. Even if we had table inheritance we'd have this problem
+  // so long as an outgoing payment could have either type of quote on them. To truly make this
+  // typesafe you might need to have seperate outgoing payment tables for each type of quote (which
+  // goes way too far)
+  if (!outgoingPayment.quote.ilpQuoteDetails) {
+    throw new Error('Missing ILP quote details')
+  }
```

Comment:
I think this is where we do an explicit lookup in `ilpQuoteDetails` table (ie, we don't fetch it in the usual `quote` request), but only fetch it when we need it here. thoughts?

That way, we don't need to mess with types at all

Reply:
Hmm... we might have to parse it either way. Either like this or explicitly looking it up. 

I guess it goes back to where this is coming from. If we were explicitly getting one type or another we could throw an error at the service level when trying to get a payment with an ilp quote and not finding it. But in this case it looks like the worker is just getting the latest payment to process (could be anything). Therefore we're going to just have to check here (or where this is called) one way or another. So I think I'd rather just have the `ilpQuoteDetails` (whether populated or `null`) on the payment by default.

Reply:
> So I think I'd rather just have the ilpQuoteDetails (whether populated or null) on the payment by default.

I would say the way that we have this set up is that the outgoing payment itself doesn't know anything about the payment method, it only just needs to have the `payment.quote.method` in order to know what to call when calling
`paymentMethodHandlerService.pay`. The only time we will need to fetch `ilpQuoteDetails` is in this ILP payment service, so we can avoid making additional requests all other times we just need to fetch the outgoing payment data

Reply:
> The only time we will need to fetch ilpQuoteDetails is in this ILP payment service, so we can avoid making additional requests all other times we just need to fetch the outgoing payment data

Makes sense... we can switch to that.

> it only just needs to have the payment.quote.method in order to know what to call when calling
paymentMethodHandlerService.pay

I don't think I'm quite following here. `isLocal` was added to the receiver to determine which method to call. Im using that to call the correct `pay`/`getQuote`. I'm not using the quote payment method for that. In fact I didn't plan on adding a new `local` method at this level as part of not presenting some additional method in the spec (just internal). Guess I'd have to look into this more but not sure if you can clarify in light of that.